File: array/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        curr, count = None, 0
        for num in nums:
            if curr is None:
                curr = num
                count = 1
            elif count == 0:
                curr = num
                count = 1
            elif curr != num:
                count -= 1
            else:
                count += 1
        return curr

File: hash-table/majority-element.rb
# @param {Integer[]} nums
# @return {Integer}
def majority_element(nums)
    a = nums[0]
    counter = 1
    1.upto(nums.size-1) do |i|
        curr = nums[i]
        if curr == a
            counter += 1
        else
            counter -= 1
            if counter == 0
                a = curr
                counter = 1
            end
        end
    end
    a
end